### PR TITLE
Fixed typos and added MacOS system files to .gitignore

### PR DESCRIPTION
**Pull request by vallieres**
Started to use gofight so I figured I might as well fix little nitpicks :P

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,32 @@ _testmain.go
...
+
+# MacOS
+
+*.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
+
```
Comment (appleboy) [inline .gitignore]: Why we need to ignore MacOS file. I think just `.DS_Store` should be included in ignore list.

Comment (vallieres) [inline .gitignore]: This is the complete list of system-generated files. For a good coverage of systems I always use this list. It originates from : https://github.com/github/gitignore/blob/master/Global/macOS.gitignore

Comment (appleboy) [inline .gitignore]: I know your point but this is the `golang` project, not macOS project. We don't need to include all macOS ignore list.

If I merge this PR. somebody want to add some ignore items like https://github.com/github/gitignore/tree/master/Global.

Comment (vallieres) [inline .gitignore]: OK, I understand I did not think of the implications of add one platform that will open the door for many more others. Let me update the PR.

**Pull request merged**